### PR TITLE
Gamified merge into dev

### PR DESCRIPTION
**Pull request by Mafusn**
## Description

This pull request includes the implementation of the following PBIs:
[App Profile] As a waste picker, I would like to be able to see and edit my profile, in order to see what personal information is connected to my profile and edit in case something is incorrect or not up to date
[App Profile] As a waste-picker, I want to have the ability to delete my account

## Changes

Delete user implemented
Implemented ability to update first name, last name and email
Unit tests made and passes
Cleanup in all touched files

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Notes for Reviewers
Tried to resolve some of the indentation changes, but was still seen as changes.

### CODE REVIEW

File: routes/userRoutes.js
Comment (TheMOHD3886) [inline routes/userRoutes.js:16]: We won't block the merge because of this, but we should have a look at the API guidelines i 
 general, here specifically: https://github.com/Educado-App/resources/blob/main/API_GUIDELINES.md#http-delete-returns-204

File: users/userList.js
Comment (TheMOHD3886) [inline users/userList.js:31]: It seems weird that we are searching for the user by first name. What if there are several users with the same first name? (Same with last name below)

File: users/userListtest.js
Comment (TheMOHD3886) [inline users/userListtest.js:1]: Move to test directory, and rename to `userList.test.js` or `userList.spec.js`.

File: users/userList.js
Comment (TheMOHD3886) [inline users/userList.js:17]: This standard seems a bit dangerous, defaulting to remove all users if no attribute is received.

**Review by TheMOHD3886 — CHANGES_REQUESTED** [review]
## Changes requested
Good work, just a few comments about error handling and such.

File: .gitignore
Comment (Mafusn) [reply .gitignore:3]: It is, but now with a star apparently

File: users/userList.js
Comment (Mafusn) [reply users/userList.js:31]: This is some old tests, and i don't think that we even use the functions in the project at all, maybe just delete them

File: users/userListtest.js
Comment (Mafusn) [reply users/userListtest.js:1]: We haven't touched those files, maybe we need to look more into them. All the tests fail when i run them

File: helpers/errorCodes.js
Comment (TheMOHD3886) [inline helpers/errorCodes.js:1]: Remember to add this change to ERRORS.md in resources and in educado-mobile repo

**Review by TheMOHD3886 — APPROVED** [review]
## Approved
Nice job, minor comments regarding endpoints and some other things

File: helpers/errorCodes.js
Comment (Mafusn) [reply helpers/errorCodes.js:1]: Done

**Pull request merged**
